float32 z_pos
